### PR TITLE
Atom buttons redesign

### PR DESCRIPTION
**Pull request by mboperator**
## Changes
- Removed traces of old button

## Screenshots
screen shot 2017-09-29 at 3 03 06 pm

### CODE REVIEW

File: src/styles/components/_button.scss [hunk 1]
```diff
@@ -1,248 +1,190 @@
...
+$button-tertiary-hover-border: 1px solid #808080;
+$button-tertiary-hover-font-color: #808080;
+
+$button-tertiary-inactive-fill: transparent;
+$button-tertiary-inactive-border: 1px solid transparent;
+$button-tertiary-inactive-font-color: #d9d9d9;
+
+.button {
+  & + .button {
+    margin: 4px;
```
Comment (brock-noah) [inline src/styles/components/_button.scss]: I feel like this is the job of a different class. A buttons wrapper that effects the layout between them.

**Comment by brock-noah** [comment]
We should target `:disabled` for them as well. Probably will be the same styles as the inactive class.

File: src/components/Button/index.jsx [hunk 2]
```diff
@@ -16,15 +16,15 @@ const Button = props => {
...
-      [styles['disabled']]: disabled,
-      [styles['bttn-lg']]: size === 'lg',
-      [styles['bttn-sm']]: size === 'sm',
-      [styles['bttn-xs']]: size === 'xs',
-      [styles['bttn-primary-solid']]: variant === 'action',
-      [styles['bttn-primary-outline']]: variant === 'secondary-action', // clarify this
+      // [styles['disabled']]: disabled,
+      [styles['button--large']]: size === 'large',
+      [styles['button--medium']]: size === 'medium',
+      [styles['button--small']]: size === 'small',
```
Comment (brock-noah) [inline src/components/Button/index.jsx]: `nb` I personally like the `xs`, `sm`, `md`, `lg`, `xl` range. Keeps things short and still clear imo

File: src/styles/components/_button.scss (hunk 1 above)
Comment (mboperator) [reply src/styles/components/_button.scss]: I agree with this -- UX does want buttons to always be separated by 4px when they're together though

Comment (brock-noah) [reply src/styles/components/_button.scss]: Separated by 4px means different things based on location. In the sidebar we want bottom margin, adding horizontal margin would add undesired space. If we want everything right aligned on a page, again undesired space.

I think we want something like this:
```
.atom-buttons-horizontal-space > .atom-button { margin-left: 4px; margin-right: 4px; }
.atom-buttons-horizontal-space > .atom-button:first-child { margin-left: 0; }
.atom-buttons-horizontal-space > .atom-button:last-child { margin-right: 0; }
```

We could rewrite the above with one selector.
```
.a-btns-h-space > .a-btn + .a-btn { margin-left: 4px; }
```
But is less better? That one is more specific for all the buttons and doesn't allow the ability to wrap a button in another element, like a tooltip. So I like the more selectors version.

File: src/components/Button/index.jsx (hunk 2 above)
Comment (remyyounes) [reply src/components/Button/index.jsx]: 👍

**Comment by brock-noah** [comment]
Hey sorry, slightly lied. `.atom-buttons-horizontal-space > .atom-button` also doesn't allow button wrappers like tooltips because of the `>`

**Pull request merged**